overops-functions/src/main/java/com/takipi/udf/microsoftteams/MicrosoftTeamsChannelFunction.java

```diff
@@ -0,0 +1,195 @@
+        } catch (Exception e) {
+            throw new IllegalArgumentException(e.getMessage(), e);
+        }
+
+        input.checkUrl();
+
+        return input;
+    }
+
+    public static void execute(String rawContextArgs, String rawInput) {
+        try {
+            executeImplementation(rawContextArgs, rawInput);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    private static void executeImplementation(String rawContextArgs, String rawInput) {
+        logUDFInput(rawContextArgs, rawInput);
+        MicrosoftTeamsInput input = getInput(rawInput);
+        ContextArgs args = (new Gson()).fromJson(rawContextArgs, ContextArgs.class);
+
+        if (!args.validate())
+            throw new IllegalArgumentException("Bad context args: " + rawContextArgs);
+        if (!args.eventValidate())
+            return;
+
+        ApiClient apiClient = args.apiClient();
+        UrlClient.Response<EventResult> eventResultResponse = getEventResultResponse(args, apiClient);
+        System.out.println("Got exceptionClassName, exception type, exceptionLocationPath, stack_frames, deployment");
+        MicrosoftTeamsUtil.EventData eventData = getEventData(apiClient, args);
+        System.out.println("Got applications, environments, deployments and servers");
+        MicrosoftTeamsUtil.TimeSlot timeSlot = getTimeSlot(MINUTES_TIME_SPAN);
+
+        SummarizedView view = getSummarizedView(rawContextArgs, args, apiClient);
+        System.out.println("Got viewId");
+
+        Graph eventsGraph = ViewUtil.getEventsGraph(apiClient, args.serviceId, view.id,
+                1, ValidationUtil.VolumeType.all, timeSlot.from, timeSlot.to);
+        System.out.println("Got machine_name, application_name, deployment_name");
+        String exceptionLinkToOverOps = EventUtil.getEventRecentLink(apiClient, args.serviceId, args.eventId,
+                timeSlot.from, timeSlot.to, eventData.applications,
+                eventData.servers, eventData.deployments);
+        System.out.println("Got ARC link");
+        MicrosoftTeamsChannelRequest microsoftTeamsChannelRequest = MicrosoftTeamsChannelRequest.newBuilder()
+                .setUrl(input.url)
+                .setEventResult(eventResultResponse.data)
+                .setExceptionLink(exceptionLinkToOverOps)
+                .setServer(eventsGraph.machine_name)
+                .setApplication(eventsGraph.application_name)
+                .setEnvironmentName(getEnvironmentName(eventData.environments, args.serviceId))
+                .setDeployment(eventsGraph.deployment_name)
+                .setDoNotAlertLink(getDoNotAlertLink(args, eventResultResponse.data.type))
+                .build();
+        UrlClient.Response<String> post = SimpleUrlClient.newBuilder().build().post(microsoftTeamsChannelRequest);
+        System.out.println("Post Microsoft Teams Webhook Channel request");
+
+        if (post.isBadResponse())
```

Comment:
Please have `{}` surrounding if code blocks, even if it's just a single line,.
In this case:

```
if (post.isBadResponse()) {
    throw new IllegalStateException("Can't send card to " + input.url);
}
```